be/src/storage/persistent_index.cpp

```diff
@@ -25,6 +72,378 @@ struct FixedKeyHash {
     }
 };
 
+static std::tuple<size_t, size_t> estimate_nshard_and_npage(size_t kv_size, size_t size, size_t usage_percent) {
+    size_t usage = size * kv_size;
+    size_t cap = usage * 100 / usage_percent;
+    size_t nshard = 1;
+    while (nshard * 1024 * 1024 < cap) {
+        nshard *= 2;
+        if (nshard == shard_max) {
+            break;
+        }
+    }
+    size_t npage = npad(cap / nshard, page_size);
+    return {nshard, npage};
+}
+
+// Page storage layout:
+//   each page has 4096 / 16 = 256 packs, ie
+//   |--------       4096 byte page             -------|
+//   |16b pack0|16b pack0| ... |16b pack254|16b pack255|
+//   | header  |       data for buckets                |
+// Header layout
+//   |BucketInfo0|BucketInfo1|...|BucketInfo14|BucketInfo15|
```

Comment:
Does every page is at 4K to improve read performance?

Reply:
yes

Reply:
4K is os page cache basic unit